src/main/clojure/clara/rules/platform.cljc

```diff
@@ -8,6 +8,22 @@
   (throw #?(:clj (IllegalArgumentException. description) :cljs (js/Error. description))))
 
 #?(:clj
+   (defn group-by-seq [f coll]
+     (let [^java.util.Map m (reduce (fn [^java.util.Map m x]
+                                      (let [k (f x)
+                                            xs (or (.get m k)
+                                                   (transient []))]
+                                        (.put m k (conj! xs x)))
+                                      m)
+                                    (java.util.LinkedHashMap.)
+                                    coll)]
+       (into []
+             (map (fn [^java.util.Map$Entry e] [(.getKey e) (persistent! (.getValue e))]))
```

Comment:
I think this will work out based on reading through http://dev.clojure.org/jira/browse/CLJ-1738 and the implementation of map at https://github.com/clojure/clojure/blob/clojure-1.8.0/src/clj/clojure/core.clj#L2618 but I wonder if just fully controlling it ourselves via a loop-recur would be safer and more resilient to any future Clojure changes.  I recall that we saw some discrepancies between Java 6 and 8 regarding this behavior - did we ever establish the reason for that?